mini-projects/01-device-driver/laptop-interface/src/model.rs

```diff
@@ -0,0 +1,194 @@
+        .unwrap();
+    view_sender
+        .send(("model".to_string(), "Start game".to_string()))
+        .unwrap();
+
+    // Reset and initialize controllers
+    let commands = vec![Command::Stop, Command::Reset, Command::Init];
+    for cmd in commands {
+        controller_sender
+            .send(("model".to_string(), cmd.to_string()))
+            .unwrap();
+    }
+    // Set LEDs
+    let commands = vec![Command::SetReadyLed, Command::SetSetLed, Command::SetGoLed];
+    for cmd in commands {
+        controller_sender
+            .send(("model".to_string(), cmd.to_string()))
+            .unwrap();
+        println!("{}", cmd.to_string());
+        std::thread::sleep(std::time::Duration::from_millis(1000));
+    }
+    // Start game
+    let commands = vec![Command::SetAllLeds, Command::Start];
+    for cmd in commands {
+        controller_sender
+            .send(("model".to_string(), cmd.to_string()))
+            .unwrap();
+    }
+}
+
+fn update_positions(
+    positions: &mut Vec<[i32; 2]>,
+    circle_state: Arc<Mutex<CircleState>>,
+    receiver: Receiver<(String, String)>,
+    view_sender: Sender<(String, String)>,
+) {
+    match receiver.try_recv() {
+        Ok((sender, message)) => {
+            if sender == "view" {
+                view_sender
+                    .send(("model".to_string(), format!("{:?}", positions)))
+                    .unwrap();
+            } else {
+                let player = sender.parse::<usize>().unwrap();
+                if message.contains("n") {
+                    positions[player][1] += 1;
+                }
+                if message.contains("nw") {
+                    positions[player][0] -= 1;
+                    positions[player][1] += 1;
+                }
+                if message.contains("w") {
+                    positions[player][0] -= 1;
+                }
+                if message.contains("sw") {
+                    positions[player][0] -= 1;
+                    positions[player][1] -= 1;
+                }
+                if message.contains("s") {
+                    positions[player][1] -= 1;
+                }
+                if message.contains("se") {
+                    positions[player][0] += 1;
+                    positions[player][1] -= 1;
+                }
+                if message.contains("e") {
+                    positions[player][0] += 1;
+                }
+                if message.contains("ne") {
+                    positions[player][0] += 1;
+                    positions[player][1] += 1;
+                }
+                if message.starts_with("adc:") {
+                    let adc_value: u16 = message[4..].parse().unwrap();
+                    let mut circle = circle_state.lock().unwrap();
+                    circle.update(positions[player][0] as u8, adc_value);
+                    println!("Player {} ADC Value: {}", player, adc_value);
+                }
```

Comment:
the ticket stated to update the amount to which the player moves by, not just the radius of a "circle" that you don't display